subsys/bluetooth/host/att.c

```diff
@@ -1549,6 +1549,23 @@ static uint8_t att_read_type_req(struct bt_att_chan *chan, struct net_buf *buf)
 		return 0;
 	}
 
+	/* Reading Database Hash is special as it may be used to make client change aware
+	 * (Core Specification 5.4 Vol 3. Part G. 2.5.2.1 Robust Caching).
+	 *
+	 * GATT client shall always use GATT Read Using Characteristic UUID sub-procedure for
+	 * reading Database Hash
+	 * (Core Specification 5.4 Vol 3. Part G. 7.3 Databse Hash)
+	 */
+	if (bt_uuid_cmp(&u.uuid, BT_UUID_GATT_DB_HASH) != 0) {
```

Comment:
Hmm, this made me think of whether we can/should reject reads to `BT_UUID_GATT_DB_HASH` from the other read requests?

Reply:
other reads (and also wrtite reqs) are already handling it (that if bt_gatt_change_aware() blocks are doing that)

Reply:
They are just blocking reading when the database is out of sync though, right? 

The core spec states that the client shall read the database sync with the read by type request (read by UUID). That means that even if the client is not out of sync, it shall not read the characteristic with e.g. a regular read request. Arguably that is a requirement on the client side, but we could (not sure if we should) block reads of that characteristic if not with read by type